app/src/main/java/com/davidtomas/taskyapp/features/auth/presentation/register/RegisterState.kt

```diff
@@ -0,0 +1,15 @@
+package com.davidtomas.taskyapp.features.auth.presentation.register
+
+import com.davidtomas.taskyapp.core.domain.util.EMPTY_STRING
+import com.davidtomas.taskyapp.core.presentation.util.UiText
+
+data class RegisterState(
+    val userName: String = String.EMPTY_STRING,
+    val userNameErrMsg: UiText = UiText.DynamicString(String.EMPTY_STRING),
+    val email: String = String.EMPTY_STRING,
+    val emailErrMsg: UiText = UiText.DynamicString(String.EMPTY_STRING),
+    val password: String = String.EMPTY_STRING,
+    val passwordErrMsg: UiText = UiText.DynamicString(String.EMPTY_STRING),
```

Comment:
Instead of making non-existent errors empty strings, I'd make them `null` instead. To me it's not obvious that just because an error message is empty there is no error. Don't be afraid to use nulls what they are intended to be used for (indicating that a value doesn't exist)